### DISCUSSION TITLE
Welcome to async_shiny discussions 💬

### DISCUSSION THREAD

**Discussion by hypebright** [hypebright/async_shiny#10] [Announcements]
## 👋 Hey there!
  We’re using Discussions as a place to connect with other R/Shiny users and share thoughts and questions that don't necessarily require opening up an issue. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other R enthusiasts.
  * Welcome others and are open-minded 💪.

 To get started, you can comment below with an introduction of yourself and tell us what you want to achieve with async programming in R and how your journey is going so far! ✨

**Comment by durraniu** [comment]
Hi, I am Umair and I am currently developing a shiny app at work for the first time. I have watched your talk and learned a lot. Thank you.

I successfully used callr package's `r_bg()` function to render a plot that calls data from a remote database. This enables the user to use other parts of the app while that plot renders. So, it seems, that everything is good for a single user. But I am not sure if I need to do anything differently if thousands of users simultaneously use the app. Could you please let me know if callr automatically takes care of thousands of users by creating a separate r session or something else need to be done?

**Reply by hypebright** [reply]
Hey Umair, great that you got `callr` working in Shiny!

It works for a single user and will work for multiple users too. If you have 5 users at the same time who trigger the background process, you will have 5 background R processes running, all with a different PID. In theory, [there's no limit](https://github.com/r-lib/processx/issues/91) (`callr` uses `processx`). The only thing is that you're not talking about 5 users, but thousands. This also means thousands of R processes will be created. And this is where you will see a bottleneck: you can only spin up as many R processes as your server or cluster allows you to. After that it will not work, and you might even completely "burn" your server 😅 , aka overload.

Long story short: it automatically takes care of it, but you need to think about when you will hit your limits

**Reply by durraniu** [reply]
Got it! Thanks for the thorough explanation.
